### ISSUE TITLE
Optional object type attribute : allow specifying the value if not present, or omit

### ISSUE THREAD

**Issue by rgarrigue** [hashicorp/terraform#30736]
### Current Terraform Version
```
1.1.0
```

### Use-cases

I'm using the optional() experimental feature to specify a configuration block for https://nats.io

```
variable "nats_leafnode_remotes" {
  type = list(object({
    url     = string
    account = optional(string) # optional's experimental
  }))
  default     = []
  description = "NATS leaf node remotes as specified in NATS configuration"
}
```

It goes through templating & jsonencode for an EC2 instance user data, here in my local just so I can output the result for a faster debug

```
locals {
  ...
  user_data = templatefile(
    "user_data.sh.tftpl",
    {
      ...
      NATS_LEAFNODE_REMOTES = jsonencode(var.nats_leafnode_remotes)
    }
  )
}
```

The user_data template look likes

```
...
echo
echo "Add NATS server configuration"
echo
cat <<EOCFG | tee /opt/nats/nats-server.conf
...
%{ if length(NATS_LEAFNODE_REMOTES) > 0 ~}
leafnodes {
    remotes = ${NATS_LEAFNODE_REMOTES}
}
%{ endif ~}
...
EOCFG
...
```

Which result in

```
leafnodes {
    remotes = [{"account":null,"url":"nats://1.2.3.4:12345"},{"account":null,"url":"nats://1.2.3.5:12345"},{"account":null,"url":"nats://1.2.3.6:12345"}]
}
```

And NATS doesn't like it. 

> Mar 24 15:44:15 ip-10-2-1-220 nats-server[7718]: nats-server: cannot find local account "null" specified in leafnode remote

If I `sed 's;null;"";g nats-server.config` it's happy. Or omitting the account key would do too.

### Attempted Solutions

Removing the attribute type fix my issue, but that defeat the point of attribute's type, and this feature request is a feedback regarding optional() to begin with

### Proposal

Idea 1, which wouldn't change the current behavior, just add to it

```
my_variable = optional(string)                            # Null if unspecified 
my_variable = optional(string,"default_value")   # Equal default_value if unspecified  
my_variable = optional(string, omit)                   # Remove the attribue entirely from the object if "omit" is specified
```

Idea 2, if my_variable is unspecified, it's removed from the object, which is a breaking change compared to actual behavior, but null can be the default value to match it

```
my_variable = optional(string)                            # Remove the attribute entirely from the object
my_variable = optional(string,"default_value")   # Equal default value if unspecified, can be "null" to match actual behavior 
```

**Comment by korinne** [comment]
Hi all, we recently released the [Terraform v1.3 alpha](https://releases.hashicorp.com/terraform/1.3.0-alpha20220608), which includes the ability to mark object type attributes as optional and set default values.

I'm following up with issues that have provided feedback on the previous experiment, and wanted to invite you all to try the alpha and provide any feedback on this updated design. You can learn more/add comments[ here](https://discuss.hashicorp.com/t/request-for-feedback-optional-object-type-attributes-with-defaults-in-v1-3-alpha/40550). Thank you so much in advance, and hope you like the feature!

**Comment by apparentlymart** [comment]
Hello again!

Terraform v1.3.0 is currently in its release candidate phase, and so its functionality is frozen now and the final release will be up next unless someone encounters a release-blocking issue with the release candidate.

The release includes support for default values essentially as described here, so I'm going to close this issue. Thanks!
